src/main/java/org/folio/services/finance/transaction/PendingPaymentWorkflowService.java

```diff
@@ -81,6 +84,25 @@ private InvoiceTransactionSummary buildInvoiceTransactionsSummary(List<InvoiceWo
       .toArray(CompletableFuture[]::new));
   }
 
+  private CompletableFuture<Void> cleanupOldEncumbrances(List<InvoiceWorkflowDataHolder> holders, RequestContext requestContext) {
+    List<String> poLineIds = holders.stream().filter(holder -> holder.getInvoiceLine() != null).map(holder -> holder.getInvoiceLine().getPoLineId()).collect(toList());
+    return encumbranceService.getEncumbrancesByPoLineIds(poLineIds, requestContext)
+      .thenAccept(transactions -> cleanupOldEncumbrances(transactions, holders, requestContext));
+  }
+
+  private void cleanupOldEncumbrances(List<Transaction> poLineTransactions, List<InvoiceWorkflowDataHolder> holders, RequestContext requestContext) {
+    for (Transaction transaction : poLineTransactions) {
+      boolean encumbranceIsNoLongerRelevant = holders.stream().noneMatch(holder -> sameFundAndPoLine(transaction, holder));
+      if (encumbranceIsNoLongerRelevant) {
+        baseTransactionService.releaseEncumbrance(transaction, requestContext);
```

Comment:
It should be in scope of async flow cleanupOldEncumbrances(...). In your case control will be returned immediately and if release failed no way to understand that